### PR TITLE
Don't recompile blacklisted path regexes

### PR DESCRIPTION
**Pull request by mjbryant**
Shaves a tiny bit of time off of processing a request. Apparently even re-compiling already compiled regexes takes time.

### CODE REVIEW

File: pyramid_zipkin/request_helper.py [hunk 1]
```diff
@@ -71,7 +71,12 @@ def should_not_sample_path(request):
     """
     blacklisted_paths = request.registry.settings.get(
         'zipkin.blacklisted_paths', [])
-    regexes = [re.compile(r) for r in blacklisted_paths]
+    # Only compile strings, since even recompiling existing
+    # compiled regexes takes time.
+    regexes = [
+        re.compile(r) if isinstance(r, str) else r
```
Comment (prat0318) [inline pyramid_zipkin/request_helper.py]: lets also handle unicode. and [use basestring](http://stackoverflow.com/a/1979107)

File: docs/source/configuring_zipkin.rst [hunk 2]
```diff
@@ -70,10 +70,12 @@ fine tune as per your use case.
...
     The `stream_name` the message will be logged to via transport layer. It defaults to `zipkin`.
 
+
 2. zipkin.blacklisted_paths
 ---------------------------
-    A list of paths as strings (accepts regex) any of which if matched with the
-    request path will not be sampled. Defaults to `[]`. Example:
+    A list of paths as strings, regex strings, or compiled regexes, any of which if matched with the
+    request path will not be sampled. Pre-compiled regexes will be the fastest. Won't behave well with
+    uncompiled unicode strings. Defaults to `[]`. Example:
```
Comment (prat0318) [inline docs/source/configuring_zipkin.rst]: curious about "Won't behave well with uncompiled unicode"

File: pyramid_zipkin/request_helper.py (hunk 1 above)
Comment (mjbryant) [inline pyramid_zipkin/request_helper.py]: basestring won't work for py3, which we support. To get this working nicely without a whole mess of if/else statements we could probably dep [six](https://pypi.python.org/pypi/six).

I thought this option of just making users `re.compile()` their unicode strings was the easiest. No one uses unicode in paths anyways.

File: docs/source/configuring_zipkin.rst (hunk 2 above)
Comment (mjbryant) [inline docs/source/configuring_zipkin.rst]: That's related to the fact that just putting u'/some_path' won't work, as I attempted to allude to in the docs.

**Comment by mjbryant** [comment]
How about this? The `six` library is dep-less (it's a single python file), and a TON of projects already dep it, so I don't think it's that big of a deal to include it. This way we handle unicode in both py2 and py3 with the same amount of code.

**Comment by prat0318** [comment]
yeah, i think it is ok to dep on `six`. waiting for travis

**Pull request merged**